src/sentry/api/endpoints/organization_events_trends.py

```diff
@@ -0,0 +1,121 @@
+            "alias": "avg_range_",
+        },
+        "user_misery": {
+            "format": "user_misery_range({}, {start}, {end}, {index})",
+            "alias": "user_misery_range_",
+        },
+    }
+
+    def has_feature(self, organization, request):
+        return features.has("organizations:internal-catchall", organization, actor=request.user)
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        with sentry_sdk.start_span(op="discover.endpoint", description="filter_params") as span:
+            span.set_tag("organization", organization)
+            try:
+                params = self.get_filter_params(request, organization)
+            except NoProjects:
+                return Response([])
+            params = self.quantize_date_params(request, params)
+
+            has_global_views = features.has(
+                "organizations:global-views", organization, actor=request.user
+            )
+            if not has_global_views and len(params.get("project_id", [])) > 1:
+                raise ParseError(detail="You cannot view events from multiple projects.")
+
+            middle = params["start"] + timedelta(
+                seconds=(params["end"] - params["start"]).total_seconds()
+                / (1 / float(request.GET.get("intervalRatio", 0.5)))
+            )
+            start, middle, end = (
+                datetime.strftime(params["start"], DateArg.date_format),
+                datetime.strftime(middle, DateArg.date_format),
+                datetime.strftime(params["end"], DateArg.date_format),
+            )
+
+        trend_function = request.GET.get("trendFunction", "p50()")
+        function, columns = parse_function(trend_function)
+        trend_column = self.trend_columns.get(function)
+        if trend_column is None:
+            return Response(
+                {"detail": "{} is not a supported trend function".format(trend_function)},
+                status=400,
+            )
+
+        selected_columns = request.GET.getlist("field")[:]
+        query = request.GET.get("query")
+        orderby = self.get_orderby(request)
+
+        with self.handle_query_errors():
+            events_results = discover.query(
+                selected_columns=selected_columns
+                + [
+                    trend_column["format"].format(*columns, start=start, end=middle, index="1"),
+                    trend_column["format"].format(*columns, start=middle, end=end, index="2"),
+                    "divide({alias}2,{alias}1)".format(alias=trend_column["alias"]),
+                    "minus({alias}2,{alias}1)".format(alias=trend_column["alias"]),
+                    "count_range({start},{end},1)".format(start=start, end=middle),
+                    "count_range({start},{end},2)".format(start=middle, end=end),
+                    "divide(count_range_2,count_range_1)",
```

Comment:
Does this use aliases that are elsewhere in the query?

Reply:
Ah yes, I should make this clearer :+1:
the `count_range({start},{end},1)` gets aliased to `count_range_1`